services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/validation/ApplicationValidator.kt

```diff
@@ -2,33 +2,87 @@ package fi.hel.haitaton.hanke.validation
 
 import fi.hel.haitaton.hanke.application.Application
 import fi.hel.haitaton.hanke.application.CableReportApplicationData
+import fi.hel.haitaton.hanke.application.Customer
+import fi.hel.haitaton.hanke.application.CustomerWithContacts
 import fi.hel.haitaton.hanke.isValidBusinessId
+import java.time.ZonedDateTime
 import javax.validation.ConstraintValidator
 import javax.validation.ConstraintValidatorContext
+import mu.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
 
 class ApplicationValidator : ConstraintValidator<ValidApplication, Application> {
-    override fun isValid(application: Application?, context: ConstraintValidatorContext?): Boolean {
-
-        val applicationData =
-            application?.applicationData
-                ?: throw InvalidApplicationDataException("Application data missing.")
-
-        return when (applicationData) {
-            is CableReportApplicationData ->
-                listOfNotNull(
-                        applicationData.customerWithContacts.customer,
-                        applicationData.contractorWithContacts.customer,
-                        applicationData.representativeWithContacts?.customer,
-                        applicationData.propertyDeveloperWithContacts?.customer,
-                    )
-                    .all {
-                        when {
-                            it.registryKey == null -> true
-                            it.registryKey.isValidBusinessId() -> true
-                            else -> throw InvalidApplicationDataException("Y-tunnus invalid.")
-                        }
-                    }
+    override fun isValid(application: Application, context: ConstraintValidatorContext?): Boolean {
+
+        return when (application.applicationData) {
+            is CableReportApplicationData -> validOrThrow(application.applicationData)
+        }
+    }
+
+    private fun validOrThrow(cableReportData: CableReportApplicationData): Boolean {
+        if (!isValidCableReport(cableReportData)) {
+            throw InvalidApplicationDataException("Received cable report contains invalid data.")
+        }
+        return true
+    }
+
+    private fun isValidCableReport(cableReportData: CableReportApplicationData): Boolean =
+        with(cableReportData) {
+            listOf(
+                    validStartAndEnd(startTime, endTime),
+                    validCustomerWithContactsNotNull(customerWithContacts),
```

Comment:
`contractorWithContacts` is also not nullable.

Reply:
Nice catch, thanks!

Reply:
I changed that to   

`validCustomersWithContacts(cableReportData.customersWithContacts())`

which is a list of not null.